lib/spack/spack/bootstrap/core.py

```diff
@@ -86,7 +89,7 @@ class Bootstrapper:
 
     config_scope_name = ""
 
-    def __init__(self, conf):
+    def __init__(self, conf: ConfigDictionary):
```

Comment:
Checked in [pep-484](https://peps.python.org/pep-0484/#the-meaning-of-annotations), you're correct:

>(Note that the return type of __init__ ought to be annotated with -> None. The reason for this is subtle. If __init__ assumed a return annotation of -> None, would that mean that an argument-less, un-annotated __init__ method should still be type-checked? Rather than leaving this ambiguous or introducing an exception to the exception, we simply say that __init__ ought to have a return annotation; the default behavior is thus the same as for other methods.)

Reply:
Done